lib/luacrypto_shash.c

```diff
@@ -26,22 +26,18 @@
 #include <lualib.h>
 #include <lauxlib.h>
 #include <lunatik.h>
+#include "luacrypto.h"
 
 LUNATIK_PRIVATECHECKER(luacrypto_shash_check, struct shash_desc *);
 
-static void luacrypto_shash_release(void *private)
+static inline void luacrypto_shash_release_tfm(struct shash_desc *obj)
 {
-	struct crypto_shash *tfm;
-	struct shash_desc *sdesc = (struct shash_desc *)private;
-
-	if (sdesc) {
-		tfm = sdesc->tfm;
-		lunatik_free(sdesc);
-		if (tfm)
-			crypto_free_shash(tfm);
-	}
+	if (obj->tfm)
+		crypto_free_shash(obj->tfm);
 }
 
+LUACRYPTO_RELEASE_CUSTOM(shash, struct shash_desc, lunatik_free, luacrypto_shash_release_tfm, obj);
```

Comment:
what's `obj` here? it's confusing.. I think a better approach is to pass only a single release callback function (which receives prive) to `LUACRYPTO_RELEASE` and define a inline for this module that ends up calling `lunatik_free()` and `luacrypto_shash_release_tfm`.. IMHO, the code will be way cleaner